---
title: "L3-Multiple-Testing"
author: "Hae Kyung Im"
date: "2020-01-13"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r}
library(tidyverse)
```


```{r functions }

fastlm = function(xx,yy)
{
  ## compute betahat (regression coef) and pvalue with Ftest
  ## for now it does not take covariates

  df1 = 2
  df0 = 1
  ind = !is.na(xx) & !is.na(yy)
  xx = xx[ind]
  yy = yy[ind]
  n = sum(ind)
  xbar = mean(xx)
  ybar = mean(yy)
  xx = xx - xbar
  yy = yy - ybar

  SXX = sum( xx^2 )
  SYY = sum( yy^2 )
  SXY = sum( xx * yy )

  betahat = SXY / SXX

  RSS1 = sum( ( yy - xx * betahat )^2 )
  RSS0 = SYY

  fstat = ( ( RSS0 - RSS1 ) / ( df1 - df0 ) )  / ( RSS1 / ( n - df1 ) )
  pval = 1 - pf(fstat, df1 = ( df1 - df0 ), df2 = ( n - df1 ))
  res = list(betahat = betahat, pval = pval)

  return(res)
}

## pvalue vs uniform

qqunif = 
function(p,BH=T,CI=T,...)
{
  p=na.omit(p)
  nn = length(p)
  xx =  -log10((1:nn)/(nn+1))
  plot( xx,  -sort(log10(p)),
     xlab=expression(Expected~~-log[10](italic(p))),
        ylab=expression(Observed~~-log[10](italic(p))),
       cex.lab=1.4,mgp=c(2,1,0),
       ... )
  abline(0,1,col='gray')
  if(BH)
    {
      abline(-log10(0.05),1, col='red',lty=1)
      abline(-log10(0.10),1, col='orange',lty=2)
      abline(-log10(0.25),1, col='yellow',lty=3)
      legend('bottomright', c("FDR = 0.05","FDR = 0.10","FDR = 0.25"),
             col=c('red','orange','yellow'),lty=1:3, cex=1)
      abline(h=-log10(0.05/nn)) ## bonferroni
    }
  if(CI)
  {
    ## create the confidence intervals
    c95 <- rep(0,nn)
    c05 <- rep(0,nn)
    ## the jth order statistic from a
    ## uniform(0,1) sample
    ## has a beta(j,n-j+1) distribution
    ## (Casella & Berger, 2002,
    ## 2nd edition, pg 230, Duxbury)
    ## this portion was posted by anonymous on
    ## http://gettinggeneticsdone.blogspot.com/2009/11/qq-plots-of-p-values-in-r-using-ggplot2.html
    
    for(i in 1:nn)
    {
      c95[i] <- qbeta(0.95,i,nn-i+1)
      c05[i] <- qbeta(0.05,i,nn-i+1)
    }

    lines(xx,-log10(c95),col='gray')
    lines(xx,-log10(c05),col='gray')
  }
}

```

## calculate probability of at least one false positive (reject null when null is true)

```{r}

alpha = 0.05

Patleastonemistake = function(m) {1 - (1-alpha)^m}

curve(Patleastonemistake,from = 1, to=100, ylab="Prob at least one wrong", xlab="m = number of tests")
grid()
abline(h=1,col='gray')


```

<!-- ## show p(null) has uniform distribution when phenotype Y is unrelated to genotype X -->

<!-- ```{r pvalue distribution} -->

<!-- nsim = 100000 -->
<!-- nsam = 100 -->
<!-- maf = 0.30 -->


<!-- Xfather = matrix( rbinom(nsam * nsim,1,maf), nsam, nsim ) -->
<!-- Xmother = matrix( rbinom(nsam * nsim,1,maf), nsam, nsim ) -->
<!-- Xboth = Xfather+ Xmother -->

<!-- Y = matrix( rnorm(nsam)) -->

<!-- pvec = rep(NA,nsim) -->
<!-- bvec = rep(NA,nsim) -->

<!-- for(ss in 1:nsim) -->
<!-- { -->
<!--   fit = fastlm(Xboth[,ss], Y) -->
<!--   pvec[ss] = fit$pval   -->
<!--   bvec[ss] = fit$betahat -->
<!-- } -->

<!-- hist(pvec,xlab="p-value",main="Histogram of p-values under Null") -->

<!-- ``` -->


## simulate a GWAS data under null (no assoc between X and Y)

```{r simulate GWAS data}

nsnp = 10000
nsam = 1000
maf = 0.30

## to simplify, we use the same maf for all SNPs in the GWAS

simGWASnull = function(nsnp,nsam,maf)
{
  Xfather = matrix( rbinom(nsam * nsnp,1,maf), nsam, nsnp )
  Xmother = matrix( rbinom(nsam * nsnp,1,maf), nsam, nsnp )
  Xboth = Xfather+ Xmother

  Y = matrix( rnorm(nsam))
  return(list(Y=Y, Xmat=Xboth))
}

simu = simGWASnull(nsnp, nsam, maf)

```

## run GWAS by regression and show that the p-values are uniformly distributed

```{r run GWAS}


pvec = rep(NA,nsnp)
bvec = rep(NA,nsnp)

for(ss in 1:nsnp)
{
  fit = fastlm(simu$X[,ss], simu$Y)
  pvec[ss] = fit$pval  
  bvec[ss] = fit$betahat
}


hist(pvec,xlab="p-value",main="Histogram of p-values under Null")

```

## show qqplot against expected null

```{r qqplot against expected null}

qqunif(pvec)

```

## what does the most significant association look like

```{r most significant assoc null}

ind = which( pvec == min(pvec) )

boxplot(simu$Y ~ simu$X[,ind])

points(jitter(simu$X[,ind]), simu$Y, type="p")

```

## example of winner's curse (even when effect size is 0, we get larger when we select significant SNPs)

```{r winners curse}

ind = which(pvec < 0.01)

df = tibble(effect = c(bvec[ind],bvec), type = c(rep("signif",length(ind)),rep("all",length(bvec)) ) )

ggplot(df, aes(abs(effect), fill=type)) + geom_density(alpha = 0.6, color=NA) + theme_bw(base_size = 15) + ggtitle("Winner's curse")

```


